packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import memoize from 'memoize-one';
+
+import { KEYCODES } from '../utils/keycodes';
+
+import styles from './Cell.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * String identifier of the section in which the Cell will be rendered.
+   */
+  sectionId: PropTypes.string.isRequired,
+  /**
+   * String identifer of the row in which the Cell will be rendered.
+   */
+  rowId: PropTypes.string.isRequired,
+  /**
+   * String identifier of the column in which the Cell will be rendered.
+   */
+  columnId: PropTypes.string.isRequired,
+  /**
+   * String-formatted width that the Cell should be rendered as. Any valid css width value is supported (i.e. 200px, 3rem).
+   */
+  width: PropTypes.string.isRequired,
+  /**
+   * Boolean indicating whether the Cell is selectable.
+   */
+  isSelectable: PropTypes.bool,
+  /**
+   * Boolean indicating whether the Cell is actively selected.
+   */
+  isSelected: PropTypes.bool,
+  /**
+   * Function that will be called upon Cell selection. The `isSelectable` prop must be true for this function to be called.
+   * Parameters: `onSelect(sectionId, rowId, columnId)`
+   */
+  onSelect: PropTypes.func,
+  /**
+   * Function that will be called upon the mouse entering the selectable region of the Cell. The `isSelectable` prop must be true for this function to be called.
+   * Parameters: `onHoverStart(event)`
+   */
+  onHoverStart: PropTypes.func,
+  /**
+   * Function that will be called upon the mouse leaving the selectable region of the Cell. The `isSelectable` prop must be true for this function to be called.
+   * Parameters: `onHoverEnd(event)`
+   */
+  onHoverEnd: PropTypes.func,
+  /**
+   * Content that will rendered within the Cell.
+   */
+  children: PropTypes.node,
+  /**
+   * Function that will be called with a ref to the Cell's selectable element. Parameters: `selectableRefCallback(selectableRef)`
+   */
+  selectableRefCallback: PropTypes.func,
+};
+
+class Cell extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleTargetClick = this.handleTargetClick.bind(this);
+
+    this.getCellStyles = memoize(this.getCellStyles);
+  }
+
+  /* eslint-disable class-methods-use-this */
+  getCellStyles(width) {
```

Comment:
What exaclty is this method doing?

Reply:
This function receives a width value and returns an Object with a width value set for the 'width' key. This object is used in the render function as an inline style declaration.

This function is also memoized in the constructor so that for a given width value, the same object reference will be returned. This allows for repeat renders of the Cell to occur more efficiently if the width value has not changed between renders.

I could define the function inline within the constructor if it would make more sense; it would at least get rid of the eslint disabling here. I just broke it out because I didn't want it "hide" the logic within the constructor.

This function isn't defined as a static or outside the class because we need a per-instance version of the memoized function. Otherwise, the memoize cache would be constantly invalidated as each Cell calls it with possibly different widths.

Reply:
should we add a description as to why its best here?